File: boilerplate/backend/shared/go/recipes/passwordless.go
package main

import (
	"fmt"
	"os"

	"github.com/supertokens/supertokens-golang/recipe/dashboard"
	"github.com/supertokens/supertokens-golang/recipe/passwordless"
	"github.com/supertokens/supertokens-golang/recipe/passwordless/plessmodels"
	"github.com/supertokens/supertokens-golang/recipe/session"
	"github.com/supertokens/supertokens-golang/recipe/userroles"
	"github.com/supertokens/supertokens-golang/supertokens"
)

func getApiDomain() string {
	apiPort := os.Getenv("VITE_APP_API_PORT")
	if apiPort == "" {
		apiPort = "3001"
	}
	apiUrl := os.Getenv("VITE_APP_API_URL")
	if apiUrl == "" {
		apiUrl = fmt.Sprintf("http://localhost:%s", apiPort)
	}
	return apiUrl
}

func getWebsiteDomain() string {
	websitePort := os.Getenv("VITE_APP_WEBSITE_PORT")
	if websitePort == "" {
		websitePort = "3000"
	}
	websiteUrl := os.Getenv("VITE_APP_WEBSITE_URL")
	if websiteUrl == "" {
		websiteUrl = fmt.Sprintf("http://localhost:%s", websitePort)
	}
	return websiteUrl
}

func getStringPointer(s string) *string {
	return &s
}

// SuperTokensConfig is the configuration for SuperTokens core with Passwordless recipe
var SuperTokensConfig = supertokens.TypeInput{
	Supertokens: &supertokens.ConnectionInfo{
		ConnectionURI: "https://try.supertokens.io",
	},
	AppInfo: supertokens.AppInfo{
		AppName:         "SuperTokens Demo App",
		APIDomain:       "http://localhost:3001",
		WebsiteDomain:   "http://localhost:3000",
		APIBasePath:     getStringPointer("/auth"),
		WebsiteBasePath: getStringPointer("/auth"),
	},
	RecipeList: []supertokens.Recipe{
		passwordless.Init(plessmodels.TypeInput{
			FlowType: "MAGIC_LINK",
			ContactMethodEmailOrPhone: plessmodels.ContactMethodEmailOrPhoneConfig{
				Enabled: true,
			},
		}),
		session.Init(nil),
		dashboard.Init(nil),
		userroles.Init(nil),
	},
} 

File: boilerplate/backend/shared/python/recipes/passwordless.py
from supertokens_python import init, InputAppInfo, SupertokensConfig
from supertokens_python.recipe import passwordless, session, dashboard, userroles
from supertokens_python.recipe.passwordless import ContactEmailOrPhoneConfig
import os

def get_api_domain() -> str:
    api_port = os.environ.get("VITE_APP_API_PORT", "3001")
    api_url = os.environ.get("VITE_APP_API_URL", f"http://localhost:{api_port}")
    return api_url

def get_website_domain() -> str:
    website_port = os.environ.get("VITE_APP_WEBSITE_PORT", "3000")
    website_url = os.environ.get("VITE_APP_WEBSITE_URL", f"http://localhost:{website_port}")
    return website_url

# Configuration for SuperTokens core and application
supertokens_config = SupertokensConfig(
    connection_uri="https://try.supertokens.com"
)

app_info = InputAppInfo(
    app_name="SuperTokens Demo App",
    api_domain=get_api_domain(),
    website_domain=get_website_domain()
)

framework = None

recipe_list = [
    passwordless.init(
        flow_type="USER_INPUT_CODE_AND_MAGIC_LINK",
        contact_config=ContactEmailOrPhoneConfig()
    ),
    session.init(),
    dashboard.init(),
    userroles.init()
] 